src/main/kotlin/org/partiql/scribe/targets/spark/SparkRewriter.kt

```diff
@@ -1,54 +1,56 @@
 package org.partiql.scribe.targets.spark
 
-import org.partiql.plan.Fn
 import org.partiql.plan.PartiQLPlan
 import org.partiql.plan.PlanNode
 import org.partiql.plan.Rel
 import org.partiql.plan.Rex
 import org.partiql.plan.Statement
-import org.partiql.plan.rel
-import org.partiql.plan.relBinding
 import org.partiql.plan.relOpProject
-import org.partiql.plan.relOpScan
-import org.partiql.plan.relType
-import org.partiql.plan.rex
-import org.partiql.plan.rexOpCallStatic
-import org.partiql.plan.rexOpLit
-import org.partiql.plan.rexOpPathKey
-import org.partiql.plan.rexOpSelect
 import org.partiql.plan.rexOpStruct
 import org.partiql.plan.rexOpStructField
 import org.partiql.plan.rexOpTupleUnion
-import org.partiql.plan.rexOpVar
 import org.partiql.plan.util.PlanRewriter
 import org.partiql.scribe.ProblemCallback
+import org.partiql.scribe.RexOpVarTypeRewriter
 import org.partiql.scribe.ScribeProblem
-import org.partiql.scribe.VarToPathRewriter
 import org.partiql.scribe.asNonNullable
-import org.partiql.scribe.expandStruct
-import org.partiql.types.CollectionType
-import org.partiql.types.StaticType
+import org.partiql.scribe.excludeBindings
+import org.partiql.types.SingleType
 import org.partiql.types.StringType
 import org.partiql.types.StructType
 import org.partiql.types.TupleConstraint
-import org.partiql.types.function.FunctionParameter
-import org.partiql.types.function.FunctionSignature
-import org.partiql.value.PartiQLValueExperimental
-import org.partiql.value.PartiQLValueType
-import org.partiql.value.stringValue
-import org.partiql.value.symbolValue
 
 /**
  * The [SparkRewriter] holds the base logic for a PartiQL plan for translation to Spark SQL.
  */
 public open class SparkRewriter(val onProblem: ProblemCallback) : PlanRewriter<Rel.Type?>() {
-
-    private val EXCLUDE_ALIAS = "\$__EXCLUDE_ALIAS__"
-
     /**
-     * Expand every wildcard including:
-     * - relation wildcards (e.g. SELECT tbl.*) and
-     * - struct wildcards (e.g. `SELECT tbl.foo.*`).
+     * For Spark, expand every wildcard, including relation wildcards (e.g. `SELECT tbl.*`) and
```

Comment:
(self-review) Additions to this file are the same as the `TrinoRewriter` and follows what was done for `RedshiftRewriter`.

Deleted portions were moved to `SparkExpandStruct.kt`.